flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/decorators/HadoopConfMountDecorator.java

```diff
@@ -0,0 +1,163 @@
+			hadoopConfVolume = new VolumeBuilder()
+				.withName(Constants.HADOOP_CONF_VOLUME)
+				.withNewConfigMap()
+					.withName(existingConfigMap.get())
+					.endConfigMap()
+				.build();
+		} else {
+			final Optional<String> localHadoopConfigurationDirectory = kubernetesParameters.getLocalHadoopConfigurationDirectory();
+			if (!localHadoopConfigurationDirectory.isPresent()) {
+				return flinkPod;
+			}
+
+			final List<File> hadoopConfigurationFileItems = getHadoopConfigurationFileItems(localHadoopConfigurationDirectory.get());
+			if (hadoopConfigurationFileItems.isEmpty()) {
+				LOG.warn("Found 0 files in directory {}, skip to mount the Hadoop Configuration ConfigMap.",
+					localHadoopConfigurationDirectory.get());
+				return flinkPod;
+			}
+
+			final List<KeyToPath> keyToPaths = hadoopConfigurationFileItems.stream()
+				.map(file -> new KeyToPathBuilder()
+					.withKey(file.getName())
+					.withPath(file.getName())
+					.build())
+				.collect(Collectors.toList());
+
+			hadoopConfVolume = new VolumeBuilder()
+				.withName(Constants.HADOOP_CONF_VOLUME)
+				.withNewConfigMap()
+					.withName(getHadoopConfConfigMapName(kubernetesParameters.getClusterId()))
+					.withItems(keyToPaths)
+					.endConfigMap()
+				.build();
+		}
+
+		final Pod podWithHadoopConf = new PodBuilder(flinkPod.getPod())
+			.editOrNewSpec()
+				.addNewVolumeLike(hadoopConfVolume)
+					.endVolume()
+				.endSpec()
+			.build();
+
+		final Container containerWithHadoopConf = new ContainerBuilder(flinkPod.getMainContainer())
+			.addNewVolumeMount()
+				.withName(Constants.HADOOP_CONF_VOLUME)
+				.withMountPath(Constants.HADOOP_CONF_DIR_IN_POD)
+				.endVolumeMount()
+			.addNewEnv()
+				.withName(Constants.ENV_HADOOP_CONF_DIR)
+				.withValue(Constants.HADOOP_CONF_DIR_IN_POD)
+				.endEnv()
+			.build();
+
+		return new FlinkPod.Builder(flinkPod)
+			.withPod(podWithHadoopConf)
+			.withMainContainer(containerWithHadoopConf)
+			.build();
+	}
+
+	@Override
+	public List<HasMetadata> buildAccompanyingKubernetesResources() throws IOException {
+		if (kubernetesParameters.getExistingHadoopConfigurationConfigMap().isPresent()) {
+			return Collections.emptyList();
+		}
+
+		final Optional<String> localHadoopConfigurationDirectory = kubernetesParameters.getLocalHadoopConfigurationDirectory();
+		if (!localHadoopConfigurationDirectory.isPresent()) {
+			return Collections.emptyList();
+		}
+
+		final List<File> hadoopConfigurationFileItems = getHadoopConfigurationFileItems(localHadoopConfigurationDirectory.get());
+		if (hadoopConfigurationFileItems.isEmpty()) {
+			LOG.warn("Found 0 files in directory {}, skip to create the Hadoop Configuration ConfigMap.", localHadoopConfigurationDirectory.get());
+			return Collections.emptyList();
+		}
+
+		final Map<String, String> data = new HashMap<>();
+		for (File file: hadoopConfigurationFileItems) {
+			data.put(file.getName(), Files.toString(file, StandardCharsets.UTF_8));
```

Comment:
`org.apache.flink.util.FileUtils` has provided two utility methods of `readFileUtf8` and `readFile` for reading file contents. I prefer to use one of them. And it would be good that I create a ticket to delete the `KubernetesUtils#getContentFromFile`?

Reply:
Both the `KubernetesUtils#getContentFromFile` and `Files.toString` in `guava18` make sense to me. If we choose latter, a `hotfix` to remove `KubernetesUtils#getContentFromFile` is enough.

@TisonKun WDYT?

Reply:
I prefer to use the existing utility methods in `FileUtils `instead of `KubernetesUtils#getContentFromFile` or `Files.toString` in guava18.

Reply:
I'd like to leverage `FileUtils.readFileUtf8` and later remove `KubernetesUtils#getContentFromFile` as it is then unused.